lib/perl/Genome/Carp.pm

```diff
@@ -0,0 +1,41 @@
+package Genome::Carp;
+
+use strict;
+use warnings;
+
+use Carp qw();
+use Exporter qw(import);
+use Sub::Install qw(install_sub);
+
+my @carp_methods = qw(carpf croakf confessf cluckf longmessf shortmessf);
+our @EXPORT_OK = (@carp_methods, 'dief', 'warnf');
+
+for my $method_name (@carp_methods) {
+    my $carp_method_name = substr($method_name, 0, -1);
+    my $carp_method = Carp->can($carp_method_name);
+    unless ($carp_method) {
+        die 'method not found: Carp::' . $carp_method_name;
+    }
+
+    install_sub({
+        code => sub {
+            my ($template, @args) = @_;
+            local $Carp::CarpLevel = 1;
```

Comment:
I don't know if I disagree but I did willfully ignore that warning.

One reason is that,

> Usually people skip too many call frames.

made me less nervous since I was setting it `1`.  Another was that the wrapper function would have to inspect `caller` which bothered me a tiny bit.